R/expect_identical_linter.R

```diff
@@ -60,18 +60,38 @@ expect_identical_linter <- function() {
   #   - skip cases like expect_equal(x, 1.02) or the constant vector version
   #     where a numeric constant indicates inexact testing is preferable
   #   - skip calls using dots (`...`); see tests
-  expect_equal_xpath <- "
+  non_integer <- "NUM_CONST[contains(text(), '.')]"
+  non_integer_negative <- glue::glue("
+    OP-MINUS
+    and count(expr) = 1
+    and expr[{non_integer}]
+  ")
+
+  expect_equal_xpath <- glue::glue("
   parent::expr[not(
       following-sibling::EQ_SUB
       or following-sibling::expr[
-        expr[1][SYMBOL_FUNCTION_CALL[text() = 'c']]
-        and expr[NUM_CONST[contains(text(), '.')]]
+        (
+          expr[1][SYMBOL_FUNCTION_CALL[text() = 'c']]
+          and expr[
+            {non_integer} or {non_integer_negative}
+          ]
+        ) or (
+          {non_integer} or {non_integer_negative}
+        ) or (
+          OP-MINUS
+          and count(expr) = 1
+          and expr[
+            expr[1][SYMBOL_FUNCTION_CALL[text() = 'c']]
+            and expr[{non_integer}]
```

Comment:
While not usual, this could also be made to catch `-c(-4.2)`, enabling a refactor of `non_integer` and reducing repetition.

Reply:
If I understood correctly your comment, it should be fixed by https://github.com/r-lib/lintr/pull/2411/commits/36a8c0c74be73b3f8303416f7d58e78a48fbe97c.